StockishDeltaV.cs

```diff
@@ -1,53 +1,68 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using KSP.UI.Screens;
+﻿﻿using KSP.UI.Screens;
 using UnityEngine;
-using PreFlightTests;
+using System.Collections;
+using UnityEngine.Collections;
 
 namespace StockishDeltaV
 {
     [KSPAddon(KSPAddon.Startup.EditorAny, false)]
     public class StockishDeltaV : MonoBehaviour
     {
-        public static StockishDeltaV instance;
-        DeltaVTest test;
+        public static StockishDeltaV _instance;
+        private DeltaVTest _test;
 
         void Awake()
         {
-            if (instance != null) Destroy(this);
-            else instance = this;
-            GameEvents.onGUIEngineersReportReady.Add(ReportReady);
-            GameEvents.onGUIEngineersReportDestroy.Add(ReportDestroyed);
-        }
+            //Actually a overridden check for UnityNull, meaning the instance could actually be destroyed
+            //e.g. destroyed by KSP, and we're instantiating a fresh one in the same tick from some other source.
+            //Checking != null would only ensure it's NOT destroyed, but we instead need to make sure it's either null or destroyed (UnityNull)
+            if (_instance == null) 
```

Comment:
TBH, I would remove the singleton semantics alltogether, then we could clean up our listeners in OnDestroy.

Reply:
The singleton stuff can go - it was put in there when I didn't really understand how the VesselDeltaV class worked. 

I thought I was going to have to create my own instance of it, and needed the DeltaVTest to be able to refer back to it.